app/packages/core/src/components/Common/utils.tsx

```diff
@@ -94,3 +94,36 @@ export const getStep = (
 
   return step;
 };
+
+export const toSlug = (name: string) => {
```

Comment:
hm... we have this method in the backend and I believe we are using it to make sure the dataset with the same name is not getting saved. I am not sure we should have this logic in the frontend and backend. If one change, the other can diverge. 🤔 

https://user-images.githubusercontent.com/109545780/217056381-6acb05c1-1990-46d7-b6c1-6db19f2ac5df.mov

Reply:
I have to double check the code but the better way could be call the api with debounce and make sure the slug is available or not (it might be doing that already). what do you think?

Reply:
yeah I had the same thoughts about wanting to reuse the backend method, but then it felt like it was making too many server requests for something that can be derived client side

Reply:
however, if we wanted to do the check on click of save view to guard against submitting the view (but not for disabling the button), it would make more sense to call the api...

Also, it may make sense to make it a rest endpoint rather than graphql query

Reply:
I think having two fairly simple functions on the backend/frontend is better than the API consequences of not having a common identifier